src/dd/dd.rs

```diff
@@ -0,0 +1,734 @@
+#![crate_name = "uu_dd"]
+
+#[macro_use]
+extern crate chan;
+extern crate chan_signal;
+
+#[macro_use]
+extern crate uucore;
+
+use std::cmp;
+use std::env;
+use std::fmt;
+use std::fs::{self, File, OpenOptions};
+use std::io::{self, Read, Seek, Write};
+use std::process::exit;
+use std::result;
+use std::sync::{Arc, Mutex};
+use std::thread;
+
+use chan_signal::Signal;
+
+const BUF_SIZE: usize = 4096;
+
+pub struct Conf {
+    inp: Option<InpFileWrap>,
+    out: Option<OutFileWrap>,
+    inp_name: String,
+    out_name: String,
+    is_stdin: bool,
+    is_stdout: bool,
+
+    inp_block_size: usize,
+    out_block_size: usize,
+    cnv_block_size: usize,
+
+    skip: usize,
+    seek: usize,
+
+    count: usize,
+
+    inp_part: usize,
+    inp_full: usize,
+    out_part: usize,
+    out_full: usize,
+    trunc: usize,
+
+    flags: u32,
+
+    short_to_short: bool,
+}
+
+impl Default for Conf {
+    fn default() -> Self {
+        Conf {
+            inp: None,
+            out: None,
+            inp_name: "".to_string(),
+            out_name: "".to_string(),
+            is_stdin: true,
+            is_stdout: true,
+
+            inp_block_size: Self::DEFAULT_BLOCK_SIZE,
+            out_block_size: Self::DEFAULT_BLOCK_SIZE,
+            cnv_block_size: 0,
+
+            skip: 0,
+            seek: 0,
+
+            count: 0,
+
+            inp_part: 0,
+            inp_full: 0,
+            out_part: 0,
+            out_full: 0,
+            trunc: 0,
+
+            flags: 0,
+
+            short_to_short: false,
+        }
+    }
+}
+
+type FlagT = u32;
+
+impl Conf {
+    const DEFAULT_BLOCK_SIZE: usize = 512;
+
+    pub fn set_if(&mut self, input_filename: &str) {
+        self.inp_name = input_filename.to_string();
+        self.is_stdin = false;
+    }
+
+    pub fn set_of(&mut self, output_filename: &str) {
+        self.out_name = output_filename.to_string();
+        self.is_stdout = false;
+    }
+
+    pub fn set_flag(&mut self, flag: FlagT) {
+        self.flags |= flag;
+    }
+
+    pub fn get_flag(flag: &str) -> FlagT {
+        match flag {
+            "block" => Conf::BLOCK,
+            "unblock" => Conf::UNBLOCK,
+            "lcase" => Conf::LCASE,
+            "ucase" => Conf::UCASE,
+            "swab" => Conf::SWAB,
+            "noerror" => Conf::NOERROR,
+            "notrunc" => Conf::NOTRUNC,
+            "sync" => Conf::SYNC,
+            _ => Conf::INVALID_FLAG,
+        }
+    }
+
+    pub fn is_flag_valid(flag: &str) -> bool {
+        flag == "ibs" || flag == "obs" || flag == "bs" || flag == "cbs" || flag == "if"
```

Comment:
This is a situation where I might use a `match` to return a `bool`.

Reply:
Fixed.